topology/probes/k8s/cache.go

```diff
@@ -157,11 +147,16 @@ func (c *ResourceCache) Stop() {
 
 // NewResourceCache returns a new cache using the associed Kubernetes
 // client and with the handler for the resource that this cache manages.
-func NewResourceCache(restClient rest.Interface, objType runtime.Object, resources string, g *graph.Graph, handler ResourceHandler) *ResourceCache {
+func NewResourceCache(restClient rest.Interface, objType runtime.Object, resources string, g *graph.Graph, handler ResourceHandler, optionalEventHandler ...*graph.EventHandler) *ResourceCache {
 	watchlist := cache.NewListWatchFromClient(restClient, resources, api.NamespaceAll, fields.Everything())
 
+	eventHandler := graph.NewEventHandler(100)
```

Comment:
If no event handler, EventHandler will be null, making NotifyEvents call segfault

Reply:
How can EventHandler be initialized to nil ?

```
        eventHandler := graph.NewEventHandler(100)
        if len(optionalEventHandler) == 1 {
                eventHandler = optionalEventHandler[0]
        }

        c := &ResourceCache{
                EventHandler:   eventHandler,
                graph:          g,
                handler:        handler,
                stopController: make(chan struct{}),
        }
```

Reply:
@hunchback My bad, it can't be nil. Still, I don't see a reason to specify `optionalEventHandlers`

Reply:
this was introduced so as to let both k8s and istio share the same eventHandler; while previously k8s probe would just pass the list of Subprobes and one could get access to the needed subprobe (which is the event listener); but in the case that we want to setup a linker between and istio resource (such as destinationrule) and a k8s resource (such as namespace) this is no longer possible ... so the solution was to use a global eventHandler (which would not be tied to the Subprobe). 

Please see k8s/namespace.go for how this is used.